app/Http/Controllers/ViewController.php

```diff
@@ -14,60 +14,34 @@ class ViewController extends BaseController
     /**
      * Files view.
      *
+     * @param Request $request
      * @param string $client
      * @param string $fileName
      *
      * @return Response
      */
     public function view(Request $request, string $client, string $fileName)
     {
-        $file = File::find($client, $fileName);
-
-        // File not found
-        if (!$file) {
-            return abort(404);
-        }
+        $file = File::findOrFail($client, $fileName);
 
         // Private file
         if ($file->visibility() !== 'public') {
-            $request->client = Client::login($request->client, $request->header('Authorization'));
-
-            if (!$request->client) {
-                return abort(403);
-            }
+            Client::loginOrFail($request->client, $request->header('x-api-key'));
         }
 
+        // Image manipulation
         if (isset($request->w) || isset($request->h)) {
             $response = Image::cache(function ($image) use ($file, $request) {
-                return $image->make($file->binary())
-                ->resize($request->w, $request->h, function ($constraint) {
-                    $constraint->aspectRatio();
-                    $constraint->upsize();
-                });
+                return $image
+                    ->make($file->binary())
+                    ->resize($request->w, $request->h, function ($constraint) {
+                        $constraint->aspectRatio();
+                        $constraint->upsize();
+                    });
             });
         }
```

Comment:
When file is private function checks if client is valid, but it doesn't check whether client has permission to a given file

Reply:
Only file owner can access private files. When you request file you must provide client and file name. So `client1` can access `/client1/test.jpg` but not `/client2/test.jpg`.

Reply:
Redundant $client and $request->client, needs to be unified